src/geode/geometry/basic_objects/triangle.cpp

```diff
@@ -47,13 +47,13 @@ namespace
     {
         try
         {
-            PivotNormalResult result;
+            absl::optional< PivotNormalResult > result;
             for( const auto pivot : geode::LRange{ 3 } )
             {
                 const auto next = pivot + 1 == 3 ? 0 : pivot + 1;
                 const geode::Vector3D edge{ points[pivot], points[next] };
-                result.lengths[pivot] = edge.length();
-                const auto edge0 = edge / result.lengths[pivot];
+                result->lengths[pivot] = edge.length();
```

Comment:
This works without initializing the object inside the optional ?

Reply:
It is initialized at optional building.